### DISCUSSION TITLE
Keep state for popout

### DISCUSSION THREAD

**Discussion by Lythenas** [caplin/FlexLayout#375] [Q&A]
If I have some react state in the tab and I float the tab, the state is lost. The same, the other way around. If I have some state in a popout and close it the state is lost.

Is there some way to share the state or restore it then the tab pops out or is moved back to the layout?

Minimal example for lost state:

```js
function TestComponent() {
  const [state, setState] = useState(false);
  return <>
    <button onClick={() => setState(x => !x)}>Toggle</button>
    <span>{state ? "true" : "false"}</span>
  </>;
}
```

Also I assume the following is impossible but asking just in case: One of my tabs contains an iframe and it is freshly loaded when popping out the tab. And then freshly loaded again when moving the tab back to the layout. Is there some way to just "move the iframe" between the popout window and the main one?

**Comment by david-trumid** [comment]
since the component is being remounted it would be better to use an atom library and store the state in a global state store
